scrabble.js

```diff
@@ -1,14 +1,121 @@
 const Scrabble = {
   score: function(word) {
-    // TODO: implement score
-  }
 
-  // TODO: add the highestScoreFrom method
+    const POINTS = {
+      A: 1, B: 3, C: 3, D: 2, E: 1, F: 4,
+      G: 2, H: 4, I: 1, J: 8, K: 5, L: 1,
+      M: 3, N: 1, O: 1, P: 3, Q: 10, R: 1,
+      S: 1, T: 1, U: 1, V: 4, W: 4, X: 8,
+      Y: 4, Z: 10
+    };
+
+    let letters = word.toUpperCase().split('');
+    let wordScore = 0;
+
+    letters.forEach(function(letter) {
+      if (!(letter in POINTS)) {
+        throw 'Word includes an invalid character.'
+      };
+      wordScore += POINTS[letter];
+    });
+
+    if (word.length > 7) {
+      throw 'Word is too long.';
+    } else if (word.length === 0) {
+      throw 'Word is too short.';
+    } else if (word.length === 7) {
+      wordScore += 50;
+    };
+
+    return wordScore;
+  },
+
+  highestScoreFrom: function(arrayOfWords) {
+    let firstWord = arrayOfWords[0];
+    let secondWord = arrayOfWords[1];
 
```

Comment:
You only look at two words here! The logic in this function looks good, but you should be doing it in a loop.

For example, this test would fail:

```javascript
it ('Handles a long list of words', function() {
  expect(Scrabble.highestScoreFrom(['a', 'e', 'i', 'eggnog', 'o', 'u'])).toBe('eggnog');
});
```